src/controllers/selectedAccount/selectedAccount.ts

```diff
@@ -83,6 +88,8 @@ export class SelectedAccountController extends EventEmitter {
   // Holds the initial load promise, so that one can wait until it completes
   initialLoadPromise: Promise<void>
 
+  cashbackStatusByAccount: CashbackStatusByAccount = {}
+
```

Comment:
This state should be private and only used to calculate what is needed for the UI. The UI doesn't have to worry about timestamps and calculate things- it only needs a banner (handled by `firstCashbackBanner`) and a function that should be called when the user clicks on the button in the banner.